src/Xamarin.Android.Tools.AndroidSdk/JdkInfo.cs

```diff
@@ -258,6 +259,12 @@ static bool AnySystemJavasInstalled ()
 					const string NameValueDelim         = " = ";
 					if (string.IsNullOrEmpty (e.Data))
 						return;
+					if (e.Data.StartsWith ("Property settings:", StringComparison.Ordinal)) {
+						foundPS = true;
+					}
+					if (!foundPS) {
+						return;
+					}
 					if (e.Data.StartsWith (ContinuedValuePrefix, StringComparison.Ordinal)) {
 						if (curKey == null)
 							throw new InvalidOperationException ($"Unknown property key for value {e.Data}!");
```

Comment:
How in general should we handle errors here? It looks like now if "Property settings:" isn't found it silently ignores the JDK, returning no properties, but if currKey is null (parsing issue) it throws an exception which (at least sometimes) apparently causes the IDE to crash. Neither mode of error handling seems ideal. Instead can it log something for errors (if possible) then return in error cases (instead of throwing) to avoid a crash?

Reply:
@BretJohnson : The fundamental question -- for which there is no answer (see also [The Error Model](http://joeduffyblog.com/2016/02/07/the-error-model/) regarding Midori?) -- is thus: how should errors be handled?  What *is* an error?

Do you want the error to be "loud and noisy", so that it's found and fixed?  Or "quiet-ish", possibly allowing it to be ignored or forgotten about?

The problem encountered here was an error.  The exception was thrown as part of asserting that something was true, and it was found to *not* be true in certain environments.  It thus failed "loud and noisy", triggering a Watson.

The benefit to this is that *I found out about the error*, and thus was able to fix it.  (Sucky for VS in the meantime?  Also, what are the semantics of a Watson?  Does it only happen if the entire IDE is torn down?  Or in less "loud and noisy" circumstances?)

The alternative is for "quiet-ish" error handling, as you describe.  This means that no Watson would have ever been reported, and likewise means we likely would have never known about this particular issue, which means it wouldn't be fixed.  (Unless these error messages are reported "somewhere", and "someone" regularly reviews the messages to try to track down errors...?)

I *personally* prefer "loud and noisy" errors -- for various definitions of "loud" and "noisy" -- because it helps ensure that my team actually gets error reports.  (Whether that be via telemetry reporting error codes, or your team appropriately forwarding Watson bugs, or...)

I don't currently see or understand how my team would otherwise know these errors even *exist*. 

That's the philosophical bit/rant.

The counter-argument is whether this code is "*actually*" important enough to warrant "loud and noisy" in the first place.  To which my answer is...I'm not actually sure.

One of the "useful" bits of `JdkInfo` is `JdkInfo.Version`, which allows *sorting* JDKs to reasonably determine which is the latest/oldest/whatever.  `JdkVersion` *tries* to use the `release` file within a JDK install, but if that file is missing, it will "fall back" to using the `java.version` property as gleamed from `java -XshowSettings:properties -version`.  I consider this to be a very important bit of functionality.

That said, does it *require* that `java -XshowSettings:properties -version` parsing be "robust"?  Not necessarily; it could instead *require* ("loud and noisy") that a `release` file exist, and *only* use that for Version information.  This might even be desirable!  I'm not entirely sure.